api/include/opentelemetry/baggage/baggage.h

```diff
@@ -274,8 +274,8 @@ class OPENTELEMETRY_EXPORT Baggage
       {
         ret.push_back(' ');
       }
-      else if (std::isalnum(str[i]) || str[i] == '-' || str[i] == '_' || str[i] == '.' ||
-               str[i] == '~')
+      else if (str[i] >= ' ' && str[i] <= '~' && str[i] != '"' && str[i] != ',' && str[i] != ';' &&
+               str[i] != '\\')
```

Comment:
Several comments:

1) Please add a comment that points to the spec: https://www.w3.org/TR/baggage/#definition, to explain what this is about.

2) Per the spec, a space (20) should be illegal. the test `str[i] >= ' '` seems broken.

3) Consider using parenthesis and comments for better readability, as in

```
if (   (str[i] >= '!' ) /* 21 */
    && (str[i] <= '~' ) /* 7E */
    && (str[i] != '"') /* 22 */
    && (str[i] != ',') /* 2C */
    && (str[i] != ';') /* 3B */
    && (str[i] != '\\')) /* 5C */
```